### ISSUE TITLE
Refactor service introspection `onboard` command to make Key Vault optional

### ISSUE THREAD

**Issue by sarath-p** [microsoft/bedrock#681]
**As a:**
Developer/Operator

**I want:**

- [x] Configure a single storage account for service introspection data and use access keys stored from Azure Key vault with a fall back option to spk config. Currently the `onboard` sets the storage key in keyvault secret, but not used while loading configuration object for `spk deployment/dashboard`.
- [x] Make --key-vault parameter optional in `onboard` so that storage access key can be configured manually in spk config file when customers are not using the Key Vault in their environment.
- [x] Create a table within the storage account when it does not exist in whether a newly created storage account or an existing storage account in `onboard` command
- [x] Return storage account output in json format from `onboard` command to help with automation. For ex: the access key in the output can be used to create a variable in Variable Group subsequently.

**So that:**

Use storage account access keys from either spk config or Azure Key Vault. 
Automate Service Introspection setup steps.

**Describe the solution you'd like:**

**Describe alternatives you've considered:**

**Additional context:**

**Does this require updates to documentation?:**

**Comment by samiyaakhtar** [comment]
This task needs to be reworded so that global config takes advantage of keyvault. The introspection commands should be decoupled from knowing the fact about whether the global config came from a file or a keyvault. To me, this task is not about refactoring dashboard and deployment functionality, it's about refactoring global config to work with keyvault. Thoughts? @andrebriggs @sarath-p

**Comment by andrebriggs** [comment]
@sarath-p these look reasonable though I think We can wait on these. The effort seems a little higher.
> Configure a single storage account for service introspection data and use access keys stored from Azure Key vault with a fall back option to spk config. Currently the onboard sets the storage key in keyvault secret, but not used by spk/dashboard config. Configuring storage access key in spk config will still be a manual step.

> Create a table within the storage account when it does not exist even when the user specified the existing storage account from onboard command

@sarath-p I think these can be a separate task that we can do more immediately for clean up:
> deployment get shows no information when no data present in table
Improvement: Showing a message "No deployments found" improves the experience instead of nothing.

> deployment get error message: When introspection is not configured, the generic message is shown "You need to run spk init to initialize." which can be improved by including details like what section/variable is missing in the configuration and inform running the spk init again. Validate method can be reused here.

The partition key is essentially the name of the AzDO project, right @samiyaakhtar ? @sarath-p can you talk more about "internalizing" it? Not sure I understand

> partition_key: The partition key can be internalized within spk/spektate since there will be only one - partition at any point of time in service intropsection based on the current implementation specified
> table_name: Table name can be internalized without a configurable option since the table is used for service introspection data

@sarath-p This also can be separate task. Effort is probably small, right?

> If below variables are used only in onboard, can they be moved to onboard parameters?
service_principal_id
service_principal_secret
subscription_id
tenant_id

**Comment by samiyaakhtar** [comment]
@andrebriggs A user may want to use partition key to separate different projects, let's say two different unrelated projects. Not making this option configurable may lead us to forcing them to not have two separate views of the introspection tool, if they would like to have that. 

As for table name, what if they're utilizing a storage account that has multiple tables in it?

**Comment by sarath-p** [comment]
@andrebriggs - please feel feel to update and/or create new issues as needed to prioritize independently.

**Comment by andrebriggs** [comment]
@sarath-p this looks good we just need to break it up into manageable pieces. Much of this work can be parallelized.

**Comment by andrebriggs** [comment]
@sarath-p what do you think the sizing of this task is now?

**Comment by andrebriggs** [comment]
Closing based on chats with @sarath-p

**Comment by andrebriggs** [comment]
Adding back based on feedback from @sarath-p . Please update
